packages/analytics-core/src/plugins/destination.ts

```diff
@@ -271,55 +255,37 @@ export class Destination implements DestinationPlugin {
     this.config.loggerProvider.warn(getResponseBodyString(res));
 
     this.config.flushQueueSize /= 2;
-
-    const tryable = this.getTryableList(list);
-    this.scheduleTryable(tryable);
+    // All the events are still queued, and will be retried with next trigger
   }
 
   handleRateLimitResponse(res: RateLimitResponse, list: Context[]) {
     const dropUserIds = Object.keys(res.body.exceededDailyQuotaUsers);
     const dropDeviceIds = Object.keys(res.body.exceededDailyQuotaDevices);
-    const throttledIndex = res.body.throttledEvents;
     const dropUserIdsSet = new Set(dropUserIds);
     const dropDeviceIdsSet = new Set(dropDeviceIds);
-    const throttledIndexSet = new Set(throttledIndex);
 
-    const retry = list.filter((context, index) => {
+    const retry = list.filter((context) => {
       if (
         (context.event.user_id && dropUserIdsSet.has(context.event.user_id)) ||
         (context.event.device_id && dropDeviceIdsSet.has(context.event.device_id))
       ) {
         this.fulfillRequest([context], res.statusCode, res.body.error);
         return;
       }
-      if (throttledIndexSet.has(index)) {
-        context.timeout = this.throttleTimeout;
-      }
+
       return true;
     });
 
     if (retry.length > 0) {
       // log intermediate event status before retry
       this.config.loggerProvider.warn(getResponseBodyString(res));
     }
-
-    const tryable = this.getTryableList(retry);
-    this.scheduleTryable(tryable);
-  }
-
-  handleOtherResponse(list: Context[]) {
-    const later = list.map((context) => {
-      context.timeout = context.attempts * this.retryTimeout;
-      return context;
-    });
-
-    const tryable = this.getTryableList(later);
-    this.scheduleTryable(tryable);
+    // All the events are still queued, and will be retried with next trigger
   }
 
   fulfillRequest(list: Context[], code: number, message: string) {
-    this.removeEvents(list);
     list.forEach((context) => context.callback(buildResult(context.event, code, message)));
+    this.removeEvents(list);
```

Comment:
Why move this after callbacks? I think it would be better to finalize the state before calling callbacks.

Reply:
That's not an intended change. Updated.